cmd/manager/main.go

```diff
@@ -674,6 +683,58 @@ func chooseAndValidateIPFamily(ipFamilyStr string, ipFamilyDefault corev1.IPFami
 	}
 }
 
+// determineSetDefaultSecurityContext determines what settings we need to use for security context by using the following rules:
+// 1. If the setDefaultSecurityContext is explicitly set to either true, or false, use this value.
+// 2. use openshift detection pulled from kubevirt
+//    https://github.com/kubevirt/kubevirt/blob/f71e9c9615a6c36178169d66814586a93ba515b5/pkg/util/cluster/cluster.go#L21
+//    to determine whether or not we are running within an openshift cluster.  If we determine we are on an openshift cluster,
+//    then default to false, otherwise, return true.
+func determineSetDefaultSecurityContext(setDefaultSecurityContext string, clientset kubernetes.Interface) (bool, error) {
+	isSet, err := strconv.ParseBool(setDefaultSecurityContext)
+	if err == nil {
+		return isSet, nil
+	}
```

Comment:
I think we should return the error here. If an unexpected value is set the operator will use the autodetect mechanism without any warning/error.

```suggestion
	if len(setDefaultSecurityContext) > 0 {
		isSet, err := strconv.ParseBool(setDefaultSecurityContext)
		if err != nil {
			return false, err
		}
		return isSet, nil
	}
```

Reply:
this is a good point.  nice catch.  👍